enzyme/test/MLIR/ProbProg/generate.mlir

```diff
@@ -0,0 +1,35 @@
+// RUN: %eopt --probprog %s | FileCheck %s
+
+module {
+  func.func private @normal(%rng : tensor<2xui64>, %mean : tensor<f64>, %stddev : tensor<f64>) -> (tensor<f64>, tensor<2xui64>, tensor<f64>, tensor<f64>)
+  func.func private @logpdf(%x : tensor<f64>, %mean : tensor<f64>, %stddev : tensor<f64>) -> tensor<f64>
+
+  func.func @test(%rng : tensor<2xui64>, %mean : tensor<f64>, %stddev : tensor<f64>) -> (tensor<f64>, tensor<2xui64>, tensor<f64>, tensor<f64>) {
+    %s:4 = enzyme.sample @normal(%rng, %mean, %stddev) { logpdf = @logpdf, symbol = #enzyme.symbol<1>, name="s", traced_output_indices = array<i64: 0>, traced_input_indices = array<i64: 1, 2>, alias_map = array<i64: 1, 0, 2, 1, 3, 2> } : (tensor<2xui64>, tensor<f64>, tensor<f64>) -> (tensor<f64>, tensor<2xui64>, tensor<f64>, tensor<f64>)
```

Comment:
per discussion, lets remove the traced_output_indiices, traced_input_indices, and alias_map, by enforcing a calling convention for pdf and logpdf fn's